client/tests/wait_resource_test.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+import featureform as ff
+from featureform.resources import TrainingSet, Feature, ResourceColumnMapping
+from featureform.serving import ServingClient
+
+from datetime import timedelta
+
+
+def wait_function_success(timeout=None):
+    time.sleep(1)
+
+def wait_function_fail(timeout=None):
+    raise ValueError("Resource took to long to return")
+    
+
+def test_wait_feature(wait_function):
+    feature = Feature(name="test_name",
+                variant="test_variant")
+    feature.wait_function = wait_function
+    try:
+        feature.wait()
+    
+
+def test_wait_label(wait_function):
+    feature = Feature(name="test_name",
+                variant="test_variant")
+    feature.wait_function = wait_function
+    try:
+        feature.wait()
+
+def test_wait_source(wait_function):
+    feature = Feature(name="test_name",
+                variant="test_variant")
+    feature.wait_function = wait_function
+    try:
+        feature.wait()
+
+def test_wait_trainingset(wait_function):
+    feature = Feature(name="test_name",
+                variant="test_variant")
+    feature.wait_function = wait_function
+    try:
+        feature.wait()
+
+
+def test_waits():
```

Comment:
this file will run all the tests then rerun them again in this function

Reply:
fixed